agent/rpc/peering/service.go

```diff
@@ -471,32 +471,16 @@ func (s *Server) Establish(
 	return resp, nil
 }
 
-// validatePeeringLocality makes sure that we don't create a peering in the cluster/partition it was generated.
-// We validate by looking at the remote PeerID from the PeeringToken and looking up that peering in the partition.
-// If there is one and the request partition is the same, then we are attempting to peer within the partition, which we shouldn't.
-// We also perform a check to verify if the ServerName of the PeeringToken overlaps with our own, we do not process it
-// unless we've been able to find the peering in the store, i.e. this peering is between two local partitions.
-func (s *Server) validatePeeringLocality(token *structs.PeeringToken, partition string) error {
-	_, peering, err := s.Backend.Store().PeeringReadByID(nil, token.PeerID)
-	if err != nil {
-		return fmt.Errorf("cannot read peering by ID: %w", err)
-	}
-
-	// If the token has the same server name as this cluster, but we can't find the peering
-	// in our store, it indicates a naming conflict.
+// validatePeeringLocality makes sure that we don't create a peering in the same cluster it was generated.
+// If the ServerName of the PeeringToken overlaps with our own, we do not accept it.
+func (s *Server) validatePeeringLocality(token *structs.PeeringToken) error {
 	serverName, _, err := s.Backend.GetTLSMaterials(false)
 	if err != nil {
 		return fmt.Errorf("failed to fetch TLS materials: %w", err)
 	}
-
-	if serverName == token.ServerName && peering == nil {
-		return fmt.Errorf("conflict - peering token's server name matches the current cluster's server name, %q, but there is no record in the database", serverName)
+	if serverName == token.ServerName {
+		return fmt.Errorf("cannot create a peering within the same cluster %q", serverName)
```

Comment:
I wonder if we could still check if `!acl.EqualPartitions(peering.GetPartition(), partition)` here and throw a different message telling operators to use service-exports for partitions instead of peering.

Reply:
That could be useful, but it adds extra complexity to the codebase with the extra query, parameter, and branching statement. I'm fine with either way. Do you think this provides enough value to users to warrant it?

Maybe I could change it to say the following always:

`cannot create a peering within the same cluster %q. Refer to the 'exported-services' config documentation if you want to export between partitions without peering.`

Reply:
I'm fine with that message, but I'm not sure how others would feel about referencing an enterprise feature in OSS logs. IMHO anything we can do now to point people to the most effective patterns without filing an issue is worth it.